extensions/ql-vscode/src/databases-ui.ts

```diff
@@ -360,6 +369,45 @@ export class DatabaseUI extends DisposableObject {
     }
   };
 
+  handleRemoveOrphanedDatabases = async (): Promise<void> => {
+    logger.log('Removing orphaned databases.');
+    let dbDirs =
+      // read directory
+      (await fs.readdir(this.storagePath, { withFileTypes: true }))
+        // remove non-directories
+        .filter(dirent => dirent.isDirectory())
+        // get the full path
+        .map(dirent => path.join(this.storagePath, dirent.name))
+        // filter databases still in workspace
+        .filter(dbDir => {
+          const dbUri = Uri.file(dbDir);
+          return this.databaseManager.databaseItems.every(item => item.databaseUri.fsPath !== dbUri.fsPath);
```

Comment:
I don't think we have to do this search for every identified directory in storage.
I suggest creating a `Set` up front containing all the `fsPath` values from `this.databaseManager.databaseItems`, and then this filter just becomes a set lookup.
My assumption here is that when this function is awaited, there is no way for the user to add databases to the workspace after the set is constructed but before cleanup completes (otherwise we have a race condition).